staging/src/k8s.io/dynamic-resource-allocation/structured/allocator.go

```diff
@@ -259,7 +259,7 @@ func (a *Allocator) Allocate(ctx context.Context, node *v1.Node) (finalResult []
 	alloc.deviceMatchesRequest = make(map[matchKey]bool)
 
 	// We can estimate the size based on what we need to allocate.
-	alloc.allocatingDevices = make(map[DeviceID]bool, minDevicesTotal)
+	alloc.allocatingDevices = make(map[DeviceID]sets.Set[int], minDevicesTotal)
```

Comment:
A plain `int` with some bit twiddling is probably more efficient here. Happy to switch to that if it makes a performance difference without becoming too opaque to understand.

Reply:
Just pushed this change. Perf evaluation TBD.